jabgui/src/main/java/org/jabref/gui/git/GitShareToGitHubDialogView.java

```diff
@@ -0,0 +1,126 @@
+package org.jabref.gui.git;
+
+import javafx.application.Platform;
+import javafx.fxml.FXML;
+import javafx.scene.control.ButtonType;
+import javafx.scene.control.CheckBox;
+import javafx.scene.control.Label;
+import javafx.scene.control.PasswordField;
+import javafx.scene.control.TextField;
+import javafx.scene.control.Tooltip;
+
+import org.jabref.gui.DialogService;
+import org.jabref.gui.StateManager;
+import org.jabref.gui.desktop.os.NativeDesktop;
+import org.jabref.gui.frame.ExternalApplicationsPreferences;
+import org.jabref.gui.preferences.GuiPreferences;
+import org.jabref.gui.util.BaseDialog;
+import org.jabref.gui.util.ControlHelper;
+import org.jabref.gui.util.IconValidationDecorator;
+import org.jabref.logic.git.prefs.GitPreferences;
+import org.jabref.logic.l10n.Localization;
+import org.jabref.logic.util.BackgroundTask;
+import org.jabref.logic.util.TaskExecutor;
+
+import com.airhacks.afterburner.views.ViewLoader;
+import de.saxsys.mvvmfx.utils.validation.visualization.ControlsFxVisualizer;
+
+public class GitShareToGitHubDialogView extends BaseDialog<Void> {
+    private static final String GITHUB_PAT_DOCS_URL =
+            "https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens";
+
+    private static final String GITHUB_NEW_REPO_URL = "https://github.com/new";
+
+    @FXML private TextField repositoryUrl;
+    @FXML private TextField username;
+    @FXML private PasswordField personalAccessToken;
+    @FXML private ButtonType shareButton;
+    @FXML private Label patHelpIcon;
+    @FXML private Tooltip patHelpTooltip;
+    @FXML private CheckBox rememberSettingsCheck;
+    @FXML private Label repoHelpIcon;
+    @FXML private Tooltip repoHelpTooltip;
+
+    private final GitShareToGitHubDialogViewModel viewModel;
+    private final DialogService dialogService;
```

Comment:
You can use @Inject here
```
    @Inject private DialogService dialogService;
    @Inject private TaskExecutor taskExecutor;
    @Inject private StateManager stateManager;
    @Inject private GuiPreferences preferences;

```
I am not yet sure about the GitPreferences  and the ExternalApplicationPreferences. 
@calixtus  Do I have to register/init the injector stuff for preferences?